src/components/application_manager/src/app_launch/app_launch_data_json.cc

```diff
@@ -255,15 +272,15 @@ bool AppLaunchDataJson::DeleteOldestAppData() {
   }
 
   // Clear Json list
-  GetSavedApplicationDataList().clear();
+  GetSavedApplicationDataList(dictionary).clear();
 
   // Copy to Json new list without oldest one
   for (std::vector<Json::Value>::iterator it = temp_json_list.begin();
        it != temp_json_list.end();
        ++it) {
```

Comment:
@YarikMamykin 
Can we use a range-based for loop like below:
for (const auto& value : temp_json_list)

Reply:
@SNiukalov fixed in https://github.com/smartdevicelink/sdl_core/pull/2626/commits/f4e32226bda388a572a9edc282e74636a66ad925